R/parser.R

```diff
@@ -459,7 +459,9 @@ token <- function(p) {
     p %then%
     space() %using%
     function(x) {
-      unlist(c(x))[2]
+      x <- x[-1]
```

Comment:
The changes to token seem to have broken the unit tests. 

```r
ex <- function(x) {
  x <- x[-1]
  x <- x[-length(x)]
  x
} 

ex1 <- function(x) {
  unlist(x)[2]
}

ex2 <- function(x) {
  x[[2]]
}

purrr::invoke_map(list(billd=ex, orig=ex1, alt=ex2), x=list(c(1), c(2), c(3)))
# $billd
# $billd[[1]]
# [1] 2
#
#
# $orig
# [1] 2
#
# $alt
# [1] 2
```

From the example above, the change you made has a different result to the original one or a better written alternative one. 

So I'm not sure if this was intentional or not - it is this part which seems to have broken the unit tests on travis.